torch/csrc/jit/fuser/compiler.cpp

```diff
@@ -329,8 +330,12 @@ std::shared_ptr<FusedKernel> compileKernel(
     }
   }
 
-  const std::string name = "kernel_" + std::to_string(next_kernel_id++);
+  // Have checked the limit at graph_fuser. Assert nothing else changing that.
+  AT_ASSERT((flat_inputs.size() + flat_outputs.size()) <=
+            fusion_kernel_args_limit);
+
```

Comment:
Now that this has come full circle, and compileKernel can no longer return nullopt, can you please change return type of this function, and remove nullopt handling logic in the executor?

Reply:
I was thinking to keep the more flexible interface able to handle the case of nullopt. But I can remove that if so desired.

Reply:
The limit check in compileKernel() has been turned to an assert, and the returned value of possible nullopt from compileKernel() has also been removed in the latest commits.